diff --git a/src/Judy.h b/src/Judy.h
index adfb5b5..031016d 100644
--- a/src/Judy.h
+++ b/src/Judy.h
@@ -30,29 +30,8 @@
 // and some tools on some platforms.
 
 
-// PLATFORM-SPECIFIC
-
-#ifdef JU_WIN /* =============================================== */
-
-typedef __int8           int8_t;
-typedef __int16          int16_t;
-typedef __int32          int32_t;
-typedef __int64          int64_t;
-
-typedef unsigned __int8  uint8_t;
-typedef unsigned __int16 uint16_t;
-typedef unsigned __int32 uint32_t;
-typedef unsigned __int64 uint64_t;
-
-#else /* ================ ! JU_WIN ============================= */
-
-// ISO C99: 7.8 Format conversion of integer types <inttypes.h>
-#include <inttypes.h>  /* if this FAILS, try #include <stdint.h> */ 
-
 // ISO C99: 7.18 Integer types uint*_t 
-//#include <stdint.h>  
-
-#endif /* ================ ! JU_WIN ============================= */
+#include <stdint.h>
 
 // ISO C99 Standard: 7.20 General utilities
 #include <stdlib.h>  
@@ -91,7 +70,7 @@ typedef void ** PPvoid_t;
 
 #ifndef _WORD_T
 #define _WORD_T
-typedef unsigned long    Word_t, * PWord_t;  // expect 32-bit or 64-bit words.
+typedef uintptr_t Word_t, * PWord_t;  // expect 32-bit or 64-bit words.
 #endif
 
 #ifndef NULL
diff --git a/src/JudyCommon/JudyPrevNextEmpty.c b/src/JudyCommon/JudyPrevNextEmpty.c
index 4da4356..e2b5791 100644
--- a/src/JudyCommon/JudyPrevNextEmpty.c
+++ b/src/JudyCommon/JudyPrevNextEmpty.c
@@ -1238,7 +1238,7 @@ LeafB1NextSubexp:	// return here to check next bitmap subexpanse.
 	    if (subexp-- > 0)		// more subexpanses.
 	    {
 		LEAFB1_STARTSUBEXP(SETLEASTDIGITS_D);
-		bitposmaskL = (1UL << (cJU_BITSPERSUBEXPL - 1));
+		bitposmaskL = (1ULL << (cJU_BITSPERSUBEXPL - 1));
 		goto LeafB1NextSubexp;
 	    }
 
diff --git a/src/JudyCommon/JudyPrivate.h b/src/JudyCommon/JudyPrivate.h
index 350631f..fb5e365 100644
--- a/src/JudyCommon/JudyPrivate.h
+++ b/src/JudyCommon/JudyPrivate.h
@@ -411,7 +411,7 @@ typedef PWord_t Pjv_t;   // pointer to JudyL value area.
 // processors.
 
 #define JU_LEASTBYTESMASK(BYTES) \
-        ((0x100UL << (cJU_BITSPERBYTE * ((BYTES) - 1))) - 1)
+        ((0x100ULL << (cJU_BITSPERBYTE * ((BYTES) - 1))) - 1)
 
 #define JU_LEASTBYTES(INDEX,BYTES)  ((INDEX) & JU_LEASTBYTESMASK(BYTES))
 
diff --git a/src/JudyCommon/JudyPrivateBranch.h b/src/JudyCommon/JudyPrivateBranch.h
index 10295ba..64de6f5 100644
--- a/src/JudyCommon/JudyPrivateBranch.h
+++ b/src/JudyCommon/JudyPrivateBranch.h
@@ -370,7 +370,7 @@ typedef struct J__UDY_BRANCH_UNCOMPRESSED
 
 // Produce 1-digit mask at specified state:
 
-#define cJU_MASKATSTATE(State)  (0xffL << (((State) - 1) * cJU_BITSPERBYTE))
+#define cJU_MASKATSTATE(State)  (0xffLL << (((State) - 1) * cJU_BITSPERBYTE))
 
 // Get byte (digit) from Index at the specified state, right justified:
 //
diff --git a/src/JudyCommon/JudyTables.c b/src/JudyCommon/JudyTables.c
index cb8b13f..4b965fe 100644
--- a/src/JudyCommon/JudyTables.c
+++ b/src/JudyCommon/JudyTables.c
@@ -260,7 +260,7 @@ FUNCTION int main()
     for (ii = 0; AllocSizes[ii] != TERMINATOR; ii++)
 	fprintf(fd," %d,", AllocSizes[ii]);
 
-    fprintf(fd," Leaf1 = %ld\";\n\n", (Word_t)cJL_LEAF1_MAXPOP1);
+    fprintf(fd," Leaf1 = %llu\";\n\n", cJL_LEAF1_MAXPOP1);
 
 #ifndef JU_64BIT
 // ================================ 32 bit ================================
